Syntax issues with the generated binding code in Swift

I used the v0.27.0 `uniffi` to generate Swift binding code, and encountered some syntax issues.

UDL file section definition:

```udl
interface DeviceManager {
    [Throws=Error]
    constructor([ByRef] string database_url);
};
```

Syntax issues code:
```swift
public convenience init(databaseUrl: String) {
    let pointer = throws try rustCallWithError(FfiConverterTypeClipError.lift) {
        uniffi_device_kit_fn_constructor_devicemanager_new(
            FfiConverterString.lower(databaseUrl), $0
        )
    }
    self.init(unsafeFromRawPointer: pointer)
}
```

Expected generated code:
```swift
public convenience init(databaseUrl: String) throws {
    let pointer = try rustCallWithError(FfiConverterTypeClipError.lift) {
        uniffi_device_kit_fn_constructor_devicemanager_new(
            FfiConverterString.lower(databaseUrl), $0
        )
    }
    self.init(unsafeFromRawPointer: pointer)
}
```

quick bisect shows that this broke in https://github.com/mozilla/uniffi-rs/pull/2048
cc @mhammond 

easy to reproduce with this patch:

```patch
--- fixtures/coverall/src/coverall.udl
+++ fixtures/coverall/src/coverall.udl
@@ -282,4 +282,9 @@ interface ISecond {
   boolean compare(IFirst? other);
 };

+interface DeviceManager {
+    [Throws=CoverallError]
+    constructor([ByRef] string database_url);
+};
+
 dictionary EmptyStruct {};
diff --git fixtures/coverall/src/lib.rs fixtures/coverall/src/lib.rs
index d7ba23856..79edd4fc1 100644
--- fixtures/coverall/src/lib.rs
+++ fixtures/coverall/src/lib.rs
@@ -584,4 +584,13 @@ impl ISecond {

 pub struct EmptyStruct;

+pub struct DeviceManager;
+
+impl DeviceManager {
+    pub fn new(url: &String) -> Result<DeviceManager, CoverallError> {
+        todo!()
+    }
+}
+
+
 uniffi::include_scaffolding!("coverall");
```

Running this:

```
UNIFFI_TESTS_DISABLE_EXTENSIONS=kts,py,rb cargo test -p uniffi-fixture-coverall
```

fails with:

```
src/uniffi-rs/target/tmp/uniffi-fixture-coverall-6e842bdd9758c583/coverall.swift:1057:19: error: expected initial value after '='
    let pointer = throws try rustCallWithError(FfiConverterTypeCoverallError.lift) {
                  ^
src/uniffi-rs/target/tmp/uniffi-fixture-coverall-6e842bdd9758c583/coverall.swift:1057:18: error: consecutive statements on a line must be separated by ';'
    let pointer = throws try rustCallWithError(FfiConverterTypeCoverallError.lift) {
                 ^
                 ;
src/uniffi-rs/target/tmp/uniffi-fixture-coverall-6e842bdd9758c583/coverall.swift:1057:19: error: expected expression
    let pointer = throws try rustCallWithError(FfiConverterTypeCoverallError.lift) {
                  ^
error: fatalError
test uniffi_foreign_language_testcase_test_coverall_swift ... FAILED
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,10 @@
 
 ## [[UnreleasedUniFFIVersion]] (backend crates: [[UnreleasedBackendVersion]]) - (_[[ReleaseDate]]_)
 
+### What's fixed?
+
+- Fixed a regression in 0.27.0 which broke throwing constructors (#2061).
+
 ### What's changed
 
 - The async runtime can be specified for constructors/methods, this will override the runtime specified at the impl block level.

--- a/fixtures/coverall/src/coverall.udl
+++ b/fixtures/coverall/src/coverall.udl
@@ -216,6 +216,14 @@ interface Patch {
     Color get_color();
 };
 
+interface FalliblePatch {
+    [Throws=CoverallError]
+    constructor();
+
+    [Throws=CoverallError, Name="secondary"]
+    constructor();
+};
+
 interface ThreadsafeCounter {
   constructor();
   void busy_wait(i32 ms);

--- a/fixtures/coverall/src/lib.rs
+++ b/fixtures/coverall/src/lib.rs
@@ -524,6 +524,18 @@ impl Patch {
     }
 }
 
+struct FalliblePatch {}
+
+impl FalliblePatch {
+    fn new() -> Result<Self> {
+        Err(CoverallError::TooManyHoles)
+    }
+
+    fn secondary() -> Result<Self> {
+        Err(CoverallError::TooManyHoles)
+    }
+}
+
 // This is a small implementation of a counter that allows waiting on one thread,
 // and counting on another thread. We use it to test that the UniFFI generated scaffolding
 // doesn't introduce unexpected locking behaviour between threads.

--- a/fixtures/coverall/tests/bindings/test_coverall.kts
+++ b/fixtures/coverall/tests/bindings/test_coverall.kts
@@ -511,6 +511,25 @@ FakeCoveralls("using_fakes_and_calling_methods_without_override_crashes").use { 
     assert(exception != null)
 }
 
+FakeCoveralls("using_fallible_constructors").use { coveralls ->
+    var exception: Throwable? = null
+    try {
+        FalliblePatch()
+    } catch (e: Throwable) {
+        exception = e
+    }
+    assert(exception != null)
+
+    exception = null;
+
+    try {
+        FalliblePatch.secondary()
+    } catch (e: Throwable) {
+        exception = e
+    }
+    assert(exception != null)
+}
+
 Coveralls("using_fakes_with_real_objects_crashes").use { coveralls ->
     val patch = FakePatch(Color.RED)
     var exception: Throwable? = null

--- a/fixtures/coverall/tests/bindings/test_coverall.py
+++ b/fixtures/coverall/tests/bindings/test_coverall.py
@@ -254,6 +254,12 @@ class TestCoverall(unittest.TestCase):
         coveralls = None
         self.assertEqual(get_num_alive(), 0)
 
+    def test_throwing_constructor(self):
+        with self.assertRaises(CoverallError.TooManyHoles):
+            FalliblePatch()
+        with self.assertRaises(CoverallError.TooManyHoles):
+            FalliblePatch.secondary()
+
     def test_bad_objects(self):
         coveralls = Coveralls("test_bad_objects")
         patch = Patch(Color.RED)

--- a/fixtures/coverall/tests/bindings/test_coverall.swift
+++ b/fixtures/coverall/tests/bindings/test_coverall.swift
@@ -274,6 +274,21 @@ do {
     assert(coveralls.getRepairs().count == 2)
 }
 
+// Test fallible constructors.
+do {
+    let _ = try FalliblePatch()
+    fatalError("ctor should have thrown")
+} catch {
+    // OK!
+}
+
+do {
+    let _ = try FalliblePatch.secondary()
+    fatalError("ctor should have thrown")
+} catch {
+    // OK!
+}
+
 // Test bytes
 do {
     let coveralls = Coveralls(name: "test_bytes")

--- a/fixtures/futures/src/lib.rs
+++ b/fixtures/futures/src/lib.rs
@@ -232,6 +232,18 @@ pub async fn say_after_with_tokio(ms: u16, who: String) -> String {
     format!("Hello, {who} (with Tokio)!")
 }
 
+#[derive(uniffi::Object)]
+pub struct FallibleMegaphone;
+
+#[uniffi::export]
+impl FallibleMegaphone {
+    // the default constructor - many bindings will not support this.
+    #[uniffi::constructor]
+    pub async fn new() -> Result<Arc<Self>, MyError> {
+        Err(MyError::Foo)
+    }
+}
+
 pub struct UdlMegaphone;
 
 impl UdlMegaphone {

--- a/fixtures/futures/tests/bindings/test_futures.swift
+++ b/fixtures/futures/tests/bindings/test_futures.swift
@@ -150,6 +150,20 @@ Task {
 	counter.leave()
 }
 
+// Test object with a fallible async ctor.
+counter.enter()
+
+Task {
+	do {
+		let _ = try await FallibleMegaphone()
+		fatalError("async ctor should have thrown")
+	} catch {
+		// OK!
+	}
+
+	counter.leave()
+}
+
 // Test foreign implemented async trait methods
 counter.enter()
 

--- a/uniffi_bindgen/src/bindings/swift/templates/macros.swift
+++ b/uniffi_bindgen/src/bindings/swift/templates/macros.swift
@@ -38,9 +38,9 @@
 {%- macro ctor_decl(callable, indent) %}
 {%- call docstring(callable, indent) %}
 public convenience init(
-    {%- call arg_list_decl(callable) -%}) {%- call async(callable) %} {
+    {%- call arg_list_decl(callable) -%}) {%- call async(callable) %} {%- call throws(callable) %} {
     {%- if callable.is_async() %}
-    let pointer = {% call throws(callable) %}
+    let pointer =
         {%- call call_async(callable) %}
         {# The async mechanism returns an already constructed self.
            We work around that by cloning the pointer from that object, then
@@ -48,8 +48,8 @@ public convenience init(
         #}
         .uniffiClonePointer()
     {%- else %}
-    let pointer = {% call throws(callable) %}
-        {%- call to_ffi_call(callable) %}
+    let pointer =
+        {% call to_ffi_call(callable) %}
     {%- endif %}
     self.init(unsafeFromRawPointer: pointer)
 }

